File: abc/abc303/a.cpp
// https://atcoder.jp/contests/abc303/submissions/42011433

#include <bits/stdc++.h>
using namespace std;

string fix_str(string s) {
  for(int i = 0; i < s.size(); i++) {
    if(s[i] == '1') s[i] = 'l';
    if(s[i] == '0') s[i] = 'o';
  }

  return s;
}

int main() {
  int n;
  cin >> n;

  string s, t;
  cin >> s >> t;
  s = fix_str(s);
  t = fix_str(t);

  cout << (s == t ? "Yes" : "No");
  return 0;
}

File: abc/abc303/a.py
# https://atcoder.jp/contests/abc303/submissions/41788527

n = int(input())
s = input()
t = input()

for i in range(n):
  if s[i] == t[i] or (s[i] in {'1', 'l'} and t[i] in {'1', 'l'} or (s[i] in {'0', 'o'} and t[i] in {'0', 'o'})): continue

  print('No')
  exit()
  
print('Yes')
